walnut/converter/converter.go

```diff
@@ -155,3 +174,36 @@ func TaskToYAML(extractedTasks []*task.Class) (err error) {
 	}
 	return
 }
+
+func RoleToYAML(input workflow.Role) (err error) {
```

Comment:
In light of the fact that you're doing filesystem operations specific to the WFT/TT configuration structure, and to avoid confusion with the other `RoleToYAML` function, I think this one could be called `GenerateWorkflowTemplate`. Similar considerations apply to `TaskToYAML`.

Generally I'd like to have the word `YAML` in functions that deal specifically with `YAML` representation, whereas higher level wrapper functions that handle filesystem and I/O aspects (as opposed to the details of YAML syntax) should have other names. It is known by design that the syntax of these files happens to be YAML, and there's no gain in making it explicit throughout the code path as long as we know what's a TT and what's a WFT. Compare this with `UnmarshalYAML` vs. `workflow.Load`, where the latter deals with a higher level of I/O and template processing, and internally calls `yaml.Unmarshal`.

Also please ensure that you only export the functions you need to export (I haven't checked if this one should be changed).

Reply:
Understood. I'll make the changes.